File: random/odd_even_car_check.py
car_no = input("Enter your car number: ")

# sum of even digits of car number
even_sum = 0
# sum of odd digits of car number
odd_sum = 0

# iterating over the car number
for i in range(len(car_no)):
    # if the digit is even
    if i % 2 == 0:
        # adding the digit to even_sum
        even_sum += int(car_no[i])
    else:
        # adding the digit to odd_sum
        odd_sum += int(car_no[i])

# checking if the sum of even digits is divisible by 4
if even_sum % 4 == 0:
    # checking if the sum of odd digits is divisible by 3
    if odd_sum % 3 == 0:
        # printing the car number is lucky
        print("Your car number is lucky")
    else:
        # printing the car number is not lucky
        print("Your car number is not lucky")

# checking if the sum of even digits is divisible by 3
elif even_sum % 3 == 0:
    # checking if the sum of odd digits is divisible by 4
    if odd_sum % 4 == 0:
        # printing the car number is lucky
        print("Your car number is lucky")
    else:
        # printing the car number is not lucky
        print("Your car number is not lucky")
        

File: random/odd_even_car_check.java
package random;
import java.util.Scanner;

public class odd_even_car_check {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        fn_odd_even(n);
        sc.close();
    }
    
    // check sum of odd digits is divisible by 3 or not
    // check sum of even digits is divisible by 4 or not

    public static void fn_odd_even(int num) {
        int sum_odd = 0;
        int sum_even = 0;
        int rem = 0;
        while(num > 0) {
            rem = num%10;
            if(rem%2 == 0) {
                sum_even += rem;
            }
            else {
                sum_odd += rem;
            }
            num = num/10;
        }
        if(sum_odd%3 == 0 || sum_even%4 == 0) {
            System.out.println("Car is Lucky");
        }
        else {
            System.out.println("Car is not Lucky");
        }
    }
}
